examples/jsm/lines/LineMaterial.js

```diff
@@ -453,247 +453,181 @@ class LineMaterial extends ShaderMaterial {
 
 		this.isLineMaterial = true;
 
-		Object.defineProperties( this, {
-
-			color: {
-
-				enumerable: true,
-
-				get: function () {
-
-					return this.uniforms.diffuse.value;
-
-				},
-
-				set: function ( value ) {
-
-					this.uniforms.diffuse.value = value;
-
-				}
-
-			},
-
-			worldUnits: {
-
-				enumerable: true,
-
-				get: function () {
-
-					return 'WORLD_UNITS' in this.defines;
-
-				},
-
-				set: function ( value ) {
-
-					if ( value === true ) {
-
-						this.defines.WORLD_UNITS = '';
-
-					} else {
-
-						delete this.defines.WORLD_UNITS;
-
-					}
-
-				}
-
-			},
-
-			linewidth: {
-
-				enumerable: true,
-
-				get: function () {
-
-					return this.uniforms.linewidth.value;
-
-				},
-
-				set: function ( value ) {
-
-					this.uniforms.linewidth.value = value;
-
-				}
-
-			},
-
-			dashed: {
-
-				enumerable: true,
+		this.setValues( parameters );
 
-				get: function () {
+	}
 
-					return Boolean( 'USE_DASH' in this.defines );
+	get color() {
 
-				},
+		return this.uniforms.diffuse.value;
 
-				set( value ) {
+	}
 
-					if ( Boolean( value ) !== Boolean( 'USE_DASH' in this.defines ) ) {
+	set color( value ) {
 
-						this.needsUpdate = true;
+		this.uniforms.diffuse.value = value;
 
-					}
+	}
 
-					if ( value === true ) {
+	get worldUnits() {
 
-						this.defines.USE_DASH = '';
+		return 'WORLD_UNITS' in this.defines;
 
-					} else {
+	}
 
-						delete this.defines.USE_DASH;
+	set worldUnits( value ) {
 
-					}
-
-				}
+		if ( value === true ) {
 
-			},
+			this.defines.WORLD_UNITS = '';
 
-			dashScale: {
+		} else {
 
-				enumerable: true,
+			delete this.defines.WORLD_UNITS;
 
-				get: function () {
+		}
 
-					return this.uniforms.dashScale.value;
+	}
 
-				},
+	get linewidth() {
 
-				set: function ( value ) {
+		return this.uniforms.linewidth.value;
 
-					this.uniforms.dashScale.value = value;
+	}
 
-				}
+	set linewidth( value ) {
 
-			},
+		if ( ! this.uniforms.linewidth ) return;
```

Comment:
Can you explain the reason for this line, please?

Reply:
Sure -- the problem is that in the `ShaderMaterial` contructor `linewidth` property is set, so the setter is called, so if `this.uniforms.linewidth` is undefined (as it is in that constructor), assign to a property of undefined fails. A cleaner solution would be to use optional chaining.

Reply:
With all due respect, I feel uneasy about the four lines I highlighted. If I had a better solution, I would suggest it.

Maybe comments would help.

@Mugen87 has good judgement about this sort of thing. I will defer to whatever he thinks.

Reply:
I think the best solution is to use optional chaining or at least do something like `if ( this.uniforms && this.uniforms.linewidth ) this.uniforms.linewidth.value = ...`.

Reply:
Since there are many contributors to three.js, it is important that we write code that can be maintained by others. IMHO, the lines I highlighted here are too hacky. It appears you agree there are better approaches.

I would be satisfied reverting to the prior implementation.